Sources/LeafKit/Exports.swift

```diff
@@ -33,6 +39,22 @@ extension Character {
     static let vertical = "|".first!
     static let underscore = "_".first!
     
+    var isHexidecimal: Bool {
+        return (.zero ... .nine).contains(self)
+            || (.A ... .F).contains(self.uppercased().first!)
+            || self == .hexNotation
+    }
+    
+    var isOctal: Bool {
+        return (.zero ... .seven).contains(self)
+        || self == .octalNotation
+    }
+    
+    var isBinary: Bool {
+        return (.zero ... .one).contains(self)
+        || self == .binaryNotation
+    }
+    
```

Comment:
Various changes to Exports are simply additional reporting observers for use during lexing